extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java

```diff
@@ -50,6 +53,11 @@ public void handle(final RoutingContext ctx) {
     }
 
     private void doHandle(final RoutingContext ctx) {
+        Instance<CurrentIdentityAssociation> association = Arc.container().select(CurrentIdentityAssociation.class);
```

Comment:
This should not be resolved per request, it should either be looked up in the constructor of cached in a volatile to improve performance.

Reply:
I'm not quite sure how to do this.
I can't do that in the constructor, because that is called from a recorder, so `Arc.container()` doesn't work yet.
If I store it into a volatile variable, I need to be able to store `null` in some cases, in which case I won't be able to say whether the association is really `null`, or just not attempted to resolve yet, therefore I'll need to keep a boolean field somewhere to distinguish this, and check it every time, so the performance improvement might not be worth it anymore?

Reply:
You should be able to depend on BeanContainerBuildItem in your processor and then just pass the container into the recorder.